sdk/src/azure/core/az_credential_client_secret.c

```diff
@@ -101,8 +112,20 @@ AZ_NODISCARD az_result az_credential_client_secret_init(
         .client_id = client_id,
         .client_secret = client_secret,
         .scopes = { 0 },
+        .authority = microsoft_aad_authority,
       },
     };
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_credential_set_authority(az_credential_client_secret* ref_credential, az_span authority)
+{
+  _az_PRECONDITION_NOT_NULL(ref_credential);
+  _az_PRECONDITION_VALID_SPAN(authority, 1, false);
+
+  ref_credential->_internal.authority = authority;
```

Comment:
Should we try to validate the contents of the user-provided custom authority in some way?

Reply:
I can't think of anything, and also I haven't seen other SDKs doing any validation.